homeassistant/components/airvisual/config_flow.py

```diff
@@ -132,15 +132,14 @@ async def _async_finish_geography(
                     LOGGER.error(err)
                     errors["base"] = "unknown"
 
-                valid_keys.add(user_input[CONF_API_KEY])
+                if errors:
+                    return self.async_show_form(
+                        step_id=error_step, data_schema=error_schema, errors=errors
```

Comment:
This looks like we can and are moving directly to the form of another step. That's not good. The form is private to its step method. We should only move to another step by awaiting that step method.

Reply:
@MartinHjelmare Absolutely. Renovating the config flow itself will make this PR bigger; would you be open to approving this one's focus on tests, then I can follow up with an additional PR to fix ^^^?